frontend/src/main/scala/bloop/engine/Dag.scala

```diff
@@ -0,0 +1,90 @@
+package bloop.engine
+
+import bloop.Project
+
+sealed trait Dag[T]
+final case class Leaf[T](value: T) extends Dag[T]
+final case class Parent[T](value: T, children: List[Dag[T]]) extends Dag[T]
+
+object Dag {
+  private[bloop] class RecursiveCycle(path: List[Project])
+      extends Exception(s"Not a DAG, cycle detected in ${path.map(_.name).mkString(" -> ")} ")
+
+  def fromMap(projectsMap: Map[String, Project]): List[Dag[Project]] = {
+    val visited = new scala.collection.mutable.HashMap[Project, Dag[Project]]()
+    val visiting = new scala.collection.mutable.LinkedHashSet[Project]()
+    val dependees = new scala.collection.mutable.HashSet[Dag[Project]]()
```

Comment:
Do you mean `dependents`? The nodes that depend on a given node? I don't think `dependee` exists

Reply:
Yup, I meant that!